build.gradle.kts

```diff
@@ -171,4 +171,41 @@ val uiTestTask = tasks.register<Test>("uiTest") {
 
 tasks.check {
     dependsOn(uiTestTask)
+}
+
+tasks.create("bumpVersion") {
+    group = "version"
+    description = "Increments the version in this build file everywhere it is used."
+    fun generateVersion(): String {
+        val updateMode = properties["mode"] ?: "patch" // By default, update the patch
+        var prjVersion = version.toString()
+        var channel = ""
+        if (prjVersion.contains("-")) {
```

Comment:
@JChrist this is a little simplistic. 

Please refer to 
* https://semver.org/#spec-item-9 and 
* https://semver.org/#spec-item-10 (see "Identifiers MUST comprise only ASCII alphanumerics and hyphens [0-9A-Za-z-]. " in particular). 

for the full details. ;) 

For example, bumping would break if we have a **valid** semver: `1.0.0+nightly-2022-08-25`, where the hyphen is not part of a pre-release identifier.

Reply:
I followed the convention for versioning/publishing the plugin, which already assumes that the channel is whatever appears after the first `-` (and does not include further hyphens)
If we really need to support build meta or multiple hyphens in release channel, then that would need to change as well